### ISSUE TITLE
Allow devDependencies to cause version bumps on dependent projects via configuration

### ISSUE THREAD

**Issue by robaca** [changesets/changesets#944]
## Problem

Like in #921 we have dev dependencies for frontend code that directly affect our services and requires their release when changed. It should not get a runtime dependency because the module itself and its dependencies are only required at build time for webpack.

We added some prepush hook to make sure the developers do not forget to always add the frontend projects to the changesets in such cases, but it would be much easier if the behavior of `changeset` in such cases would be configurable per project.

## Proposed solution

Allow specifying a list of project names in `.changeset/config.json` that should still cause version bumps on dependent projects even when being a devDependency.

**Comment by Andarist** [comment]
How would you call this option and what shape would it have?

**Comment by robaca** [comment]
I don't know how other tools handle this. I would expect something like

```json
{
  "bumpOnDevDependencies: ["@myscope/shared", "@myscope/frontend", "@myscope/react-components"]
}
```

The changelogs should then also contain these dependency updates and be able to be rewritten via a changelog generator.

**Comment by mrmckeb** [comment]
@Andarist would you accept a PR for this?

We have a technical reason that we require this too:
- Published packages depends on unpublished monorepo packages.
- The unpublished packages are bundled (there are a few reasons for this), and are listed as `devDependencies`.

We have thought about using `fixed` or `linked`, but that would require overhead too, and adds a lot of repo noise.

**Comment by Andarist** [comment]
@mrmckeb yes, a PR sounds good. I think that a good start would be adding a top-level option as suggested by @robaca . We could then bikeshed the name and the exact semantics at the code review stage.

**Comment by mrmckeb** [comment]
Thanks! I've put this PR together as an MVP, with tests:
https://github.com/changesets/changesets/pull/1159

**Comment by trystan2k** [comment]
Hi. We would also be very interested in this change. 
Do you have an ETA to have the PR reviewed ?
Thanks (special to @mrmckeb for the PR !)

**Comment by mrmckeb** [comment]
I'm still waiting on feedback on that PR sorry, I'm not a member of this project and can't merge - just a first-time contributor! 🙂

**Comment by eckdanny** [comment]
I too need something like this. Scenario:

We generate an `importmap` that includes CDN paths to all the released artifacts (which is itself published). The `importmap` is just a json manifest, so obviously it doesn't have a runtime dependency on anything, but I want it to be patch-bumped when any of the referenced pkgs (identified via `devDependencies`[?]) is rev'd like is being proposed.

I don't love the idea (has potential to create cyclic dependencies) of using the <samp>[dev|optional]dependencies</samp> member to identify those _"is related to, but doesn't really depend on"_ relationships. I thought maybe there's something in the docs like a hook or a configuration option or something, and I found this issue.

**Comment by manuel3108** [comment]
Is there any chance this can be reviewed and merged? We are always limited by #1011 or this. Merging one of those two PR would solve so many problems for us.
